extensions/bootstrap/ButtonDropdown.php

```diff
@@ -79,11 +79,13 @@ public function run()
         $options = $this->containerOptions;
         $tag = ArrayHelper::remove($options, 'tag', 'div');
 
-        echo Html::beginTag($tag, $this->containerOptions);
-        echo "\n" . $this->renderButton();
-        echo "\n" . $this->renderDropdown();
-        echo "\n" . Html::endTag($tag);
         $this->registerPlugin('button');
+        return implode("\n", [
```

Comment:
Ob_start makes echo faster not slower. That's why it's enabled by default
in recent php versions.

Nonetheless line endings are unneeded but readable HTML doesn't hurt.

Same for implode vs string concatenation, readable php is worth something
IMO.
On Feb 26, 2015 12:05 AM, "Alexander Makarov" <notifications@github.com>
wrote:

> In extensions/bootstrap/ButtonDropdown.php
> <https://github.com/yiisoft/yii2/pull/7419#discussion_r25389252>:
>
> >          $this->registerPlugin('button');
> > +        return implode("\n", [
>
> Currently not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yiisoft/yii2/pull/7419/files#r25389252>.
>